comunication.py

```diff
@@ -0,0 +1,156 @@
+    return na,nb,nc,nk
+
+def main():
+    cmd = -1
+    # sep, linhas ignorar, alg sig, max iterac, dif erro
+    try:
+        with open('config.txt','r') as f:
+            config = f.readline()
+            config = config.split('-')
+            for i in range(4):
+                config[i+1] = config[i+1]
+            config = config[:-1]
+            
+    except:
+        config = [",",1,4,100,0.05]
+    filename = None
+    repeat_file = False
+    # print(config)
+    while(cmd != 0):
+        cmd = initial_menu()
+        if cmd <= 3 and cmd != 0:
+            #Abrir o seletor de arquivos
+            if not repeat_file:
+                file = False
+                input("Precione ENTER para escolher um arquivo ")
+                filename = askopenfilename() # show an "Open" dialog box and return the path to the selected file
+                if filename is not None:
+                    if filename.endswith('.csv') or filename.endswith('.txt'):
+                        data = load_data(filename,delim=config[0],skip_rows=int(config[1]))
+                        nu = int(input("Informe quantas colunas de entradas há na amostragem:\n>> "))
+                        if nu > data.shape[1]-1:
+                            print("***Valor inválido, não há colunas referentes a saída***\n")
+                        else:
+                            file = True #ok, tudo certo com o file
+                            ny = data.shape[1]-nu
+                            u,y = sep_data(nu,ny,data)
+            if file:
+                na, nb, nc, nk = get_order_polys(cmd)
+                if   cmd == 1:
+                    m = solve.mq_interface(na,nb,nk,u,y,prec=int(config[2]))
+                    if nu == 1 and ny == 1:
+                        p = input("Deseja plotar os dados?[Y/N]\n>> ")
+                        if p == 'y' or p == 'Y':
+                            plot.plot(m,u,y)
+                elif cmd == 2:
+                    m = solve.mqe_interface(na,nb,nc,nk,u,y,float(config[4])/100,int(config[3]),int(config[2]))
+                    if nu == 1 and ny == 1:
+                        p = input("Deseja plotar os dados?[Y/N]\n>> ")
+                        if p == 'y' or p == 'Y':
+                            plot.plot(m,u,y)
+                elif cmd == 3:
+                    solve.mqr_interface(na,nb,nk,u,y,int(config[2]))
+
+                print("\n-------------------\n")
+                repeat = input("Deseja usar os mesmos dados?[Y/N]\n>> ")
+                if repeat == 'Y' or repeat == 'y':
+                    repeat_file = True
+                else:
+                    repeat_file = False    
+        elif cmd == 4:
+            print_infos()
+        elif cmd == 5:
+           config = change_config(print_config_menu(config),config)
+           with open('config.txt','w') as f:
+               for item in config:
+                   f.write(str(item)+"-")
+           # print(config)
+
+main()
```

Comment:
@nycolasRamires if you want to run this as a script, you need to modify this to the following:

`if __name__ == __main__:
     main()`

Otherwise, when someone loads the entire module, it will always run main()